src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.cs

```diff
@@ -886,7 +904,21 @@ private void DeleteButtonClick()
 
 		partial void OnDeleteButtonClickPartial();
 
-		internal void OnSelectionChanged() => SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
+		internal void OnSelectionChanged()
+		{
+#if __WASM__
+			if (Text is { Length: > 0 } text || Text.TryGetStringByRange(SelectionStart, SelectionLength, out _))
+			{
+				SelectionChanged?.Invoke(this, new RoutedEventArgs(this));
+			}
+			else
+			{
+				this.ResetCarret();
+			}
```

Comment:
It'd be best to move this code to the TextBox.wasm.cs file, and use a partial method to call it.

Reply:
It was fixed.